Orange/widgets/data/tests/test_owtranspose.py

```diff
@@ -13,27 +15,130 @@ def setUp(self):
-        self.send_signal(self.widget.Inputs.data, None)
-        self.assertEqual(self.widget.data, None)
-        self.assertIsNone(self.get_output(self.widget.Outputs.data))
+        self.send_signal(widget.Inputs.data, None)
+        self.assertIsNone(self.get_output(widget.Outputs.data))
+
+    def test_feature_type(self):
+        widget = self.widget
+        data = Table("conferences.tab")
+        metas = data.domain.metas
+        domain = data.domain
+        # Put one non-string column to metas, so widget must skip it
+        domain2 = Domain(domain.attributes[:-1],
+                         domain.class_vars,
+                         (domain.attributes[0], ) + domain.metas)
+        data2 = Table(domain2, data)
+
+        widget.feature_type = widget.GENERIC
+        self.send_signal(widget.Inputs.data, data2)
+
+        # By default, the widget switches from GENERIC to the first string meta
+        self.assertEqual(widget.feature_type, widget.FROM_META_ATTR)
+        self.assertIs(widget.feature_names_column, metas[0])
+        output = self.get_output(widget.Outputs.data)
+        self.assertListEqual(
+            [a.name for a in output.domain.attributes],
+            [metas[0].to_val(m) for m in data.metas[:, 0]])
 
-    def test_parameters(self):
-        """Check widget's output for all possible values of parameters"""
-        self.send_signal(self.widget.Inputs.data, self.zoo)
+        # Test that the widget takes the correct column
+        widget.feature_names_column = metas[1]
+        widget.apply()
+        output = self.get_output(widget.Outputs.data)
         self.assertListEqual(
-            [a.name for a in self.get_output(self.widget.Outputs.data).domain.attributes],
-            [self.zoo.domain.metas[0].to_val(m) for m in self.zoo.metas[:, 0]])
-        self.widget.feature_radio.buttons[0].click()
-        self.widget.apply()
-        self.assertTrue(all(["Feature" in x.name for x
-                             in self.get_output(self.widget.Outputs.data).domain.attributes]))
+            [a.name for a in output.domain.attributes],
+            [metas[1].to_val(m) for m in data.metas[:, 1]])
+
+        # Switch to generic
+        widget.feature_type = widget.GENERIC
+        widget.apply()
+        output = self.get_output(widget.Outputs.data)
+        self.assertTrue(
+            all(x.name.startswith("Feature ")
+                for x in output.domain.attributes))
+
+        # Check that the widget uses the supplied name
+        widget.feature_name = "Foo"
+        widget.apply()
+        output = self.get_output(widget.Outputs.data)
+        self.assertTrue(
+            all(x.name.startswith("Foo ") for x in output.domain.attributes))
+
+        # Check that the widget uses default when name is not given
+        widget.feature_name = ""
+        widget.apply()
+        output = self.get_output(widget.Outputs.data)
+        self.assertTrue(
+            all(x.name.startswith("Feature ")
```

Comment:
Wouldn't it be better to test that names start with `OWTranspose.DEFAULT_PREFIX`?
(in case someone changes the default in the future)